SimCalorimetry/HGCalSimProducers/plugins/HGCDigitizerPluginFactory.cc

```diff
@@ -0,0 +1,11 @@
+#include "SimCalorimetry/HGCalSimProducers/interface/HGCDigitizerPluginFactory.h"
+#include "HGCEEDigitizer.h"
+#include "HGCHEfrontDigitizer.h"
+#include "HGCHEbackDigitizer.h"
+#include "HFNoseDigitizer.h"
+
+EDM_REGISTER_PLUGINFACTORY(HGCDigitizerPluginFactory, "HGCDigitizerPluginFactory");
+DEFINE_EDM_PLUGIN(HGCDigitizerPluginFactory, HGCEEDigitizer, "HGCEEDigitizer");
```

Comment:
The use of a factory to handle the per-subdetector digitizers is very nice.

Since these classes are now moved into the plugins directory, the content of the header files could put into the .cc files, as we usually recommend for plugins. Then each plugin could be registered in its own .cc file.

One could also define a simpler macro:
```cpp
#define DEFINE_HGC_DIGITIZER(name) DEFINE_EDM_PLUGIN(HGCDigitizerPluginFactory, name, #name);
```